### PR TITLE
Inject mounter into volume plugins

### PR DESCRIPTION
**Pull request by pmorie**
Split out from #6936; inject mounter into volume plugins

### CODE REVIEW

**Comment by pmorie** [comment]
@vmarmol PTAL

File: pkg/volume/empty_dir/empty_dir.go [hunk 1]
```diff
@@ -238,6 +230,7 @@ func (ed *emptyDir) TearDownAt(dir string) error {
 	if err != nil {
 		return err
 	}
+	glog.Infof("TearDownAt: %v medium: %v isMnt: %v", dir, medium, isMnt)
```
Comment (vmarmol) [inline pkg/volume/empty_dir/empty_dir.go]: Remove?

**Comment by vmarmol** [comment]
LGTM, just the one nit.

**Comment by pmorie** [comment]
@vmarmol Fixed

**Comment by vmarmol** [comment]
LGTM

File: pkg/volume/aws_ebs/aws_ebs.go [hunk 2]
```diff
@@ -68,9 +68,9 @@ func (plugin *awsElasticBlockStorePlugin) GetAccessModes() []api.AccessModeType
 	}
 }
 
-func (plugin *awsElasticBlockStorePlugin) NewBuilder(spec *volume.Spec, podRef *api.ObjectReference, _ volume.VolumeOptions) (volume.Builder, error) {
+func (plugin *awsElasticBlockStorePlugin) NewBuilder(spec *volume.Spec, podRef *api.ObjectReference, _ volume.VolumeOptions, mounter mount.Interface) (volume.Builder, error) {
```
Comment (thockin) [inline pkg/volume/aws_ebs/aws_ebs.go]: At this point there is almost no reason for newBuilderInternal() in most volumes.  We should flatten those and tets through NewBuilder.

Comment (pmorie) [inline pkg/volume/aws_ebs/aws_ebs.go]: Good point

**Pull request merged**